topics/MBC_cluster_conversion/mbc_divider_script.py

```diff
@@ -131,6 +133,44 @@ def extract_digits(s):
     return "".join(filter(str.isdigit, s))
 
 
+def safer_drive_scan(drives_container_host_id, old_host_id):
+    retries = 15
+    drives_list = json.loads(run_shell_command('/bin/sh -c "weka cluster drive --host {} -J"'.format(old_host_id)))
+    drives_uuids = [disk["uuid"] for disk in drives_list]
+    print(drives_uuids)
+    scan_disk_command = '/bin/sh -c "weka cluster drive scan {}"'.format(drives_container_host_id)
+    run_shell_command(scan_disk_command)
+    sleep(2)
+    for i in range(retries):
+        drives_list = json.loads(run_shell_command('/bin/sh -c "weka cluster drive --host {} -J"'.format(old_host_id)))
+        drives_uuids = [disk["uuid"] for disk in drives_list]
+        if not drives_uuids:
+            return
+    logger.warning("the following drives did not move to the new container, {} retrying".format(drives_uuids))
+    raise
+
+
+def check_and_fix_machine_identifier_in_failure_domain(host_id):
+    host_row = json.loads(run_shell_command('/bin/sh -c "weka cluster host {} -J"'.format(host_id)))[0]
+    failure_domain_machine_id = json.loads(run_shell_command(
+        '/bin/sh -c "weka debug config show failureDomains[{}].machineIdentifier"'
+        .format(extract_digits(host_row['failure_domain_id']))))
+    host_row_machine_id = json.loads(run_shell_command(
+        '/bin/sh -c "weka debug config show hosts[{}].machineIdentifier"'.format(host_id)))
+    if failure_domain_machine_id != host_row_machine_id:
+        logger.info('Fixing the machine identifier in {} to fit host {}'.format(host_row['failure_domain_id'], host_id))
```

Comment:
`logger.info` supports formatting syntax and passing as parameters of all parameters.